controllers/kafkamanagement/kafkauser_controller.go

```diff
@@ -79,114 +75,254 @@ func (r *KafkaUserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		l.Error(err, "Unable to fetch Kafka User", "request", req)
 		return models.ReconcileRequeue, nil
 	}
+	secret := &v1.Secret{}
+	kafkaUserSecretNamespacedName := types.NamespacedName{
+		Name:      user.Spec.SecretRef.Name,
+		Namespace: user.Spec.SecretRef.Namespace,
+	}
+	err = r.Client.Get(ctx, kafkaUserSecretNamespacedName, secret)
+	if k8serrors.IsNotFound(err) {
+		l.Error(err, "Cannot get Kafka User credentials. Secret is not found",
+			"request", req,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.FetchFailed,
+			"Fetch user credentials secret is failed. Reason: %v",
+			err,
+		)
+		return models.ReconcileRequeue, nil
+	}
 
-	switch user.Annotations[models.ResourceStateAnnotation] {
-	case models.CreatingEvent:
-		return r.handleCreateKafkaUser(ctx, user, l), nil
-
-	case models.UpdatingEvent:
-		return r.handleUpdateKafkaUser(ctx, user, l), nil
+	username, password, err := r.getKafkaUserCredsFromSecret(user.Spec)
```

Comment:
We already have the `getUserCreds` func that returns a user's credentials from the secret. Please use it.

Reply:
Kafkauser needs his own method